discovery-provider/src/challenges/challenge.py

```diff
@@ -230,14 +230,28 @@ def process(self, session, event_type: str, event_metadatas: List[EventMetadata]
             self._updater.on_after_challenge_creation(session, to_create_metadata)
 
             # Update all the challenges
-
             in_progress_challenges = [
                 challenge
                 for challenge in existing_user_challenges
                 if not challenge.is_complete
             ]
+
             to_update = in_progress_challenges + new_user_challenges
 
+            # Filter out challenges for deactivated users
```

Comment:
I think this is very close to the right spot, but it looks like we can call on_after_challenge_creation with UserChallenges that won't actually be made (bc they're filtered out below). I think this would actually cause some issues, because if a user ever reactivates, we would try to call on_after_challenge_creation a second time, which might fail (since that method often creates secondary rows in the DB to track other state). Think we have to move this up a bit to prevent any to-be-filtered challenges from being passed into that method